src/platform/plugins/private/presentation_panel/public/panel_actions/show_config_panel_action/show_config_panel_action.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { i18n } from '@kbn/i18n';
+
+import {
+  EmbeddableApiContext,
+  CanAccessViewMode,
+  apiCanAccessViewMode,
+  getInheritedViewMode,
+  getViewModeSubject,
+  HasReadOnlyCapabilities,
+  hasReadOnlyCapabilities,
+} from '@kbn/presentation-publishing';
+import {
+  Action,
+  FrequentCompatibilityChangeAction,
+  IncompatibleActionError,
+} from '@kbn/ui-actions-plugin/public';
+import { ACTION_SHOW_CONFIG_PANEL } from './constants';
+
+export type ShowConfigPanelActionApi = CanAccessViewMode & HasReadOnlyCapabilities;
+
+const isApiCompatible = (api: unknown | null): api is ShowConfigPanelActionApi => {
+  return hasReadOnlyCapabilities(api) && apiCanAccessViewMode(api);
+};
+
+export class ShowConfigPanelAction
+  implements Action<EmbeddableApiContext>, FrequentCompatibilityChangeAction<EmbeddableApiContext>
+{
+  public readonly type = ACTION_SHOW_CONFIG_PANEL;
+  public readonly id = ACTION_SHOW_CONFIG_PANEL;
+  public order = 50;
+
+  constructor() {}
+
+  public getDisplayName({ embeddable }: EmbeddableApiContext) {
+    if (!isApiCompatible(embeddable)) throw new IncompatibleActionError();
```

Comment:
nit: isApiCompatible check is not needed since `getDisplayName` does not use action context.

Reply:
Based on @MichaelMarcialis feedback I've leveraged it now to provide dynamic labelling